bundles/com.salesforce.bazel.eclipse.core/src/com/salesforce/bazel/eclipse/core/model/discovery/JavaAspectsClasspathInfo.java

```diff
@@ -373,11 +404,33 @@ public Collection<ClasspathEntry> compute() throws CoreException {
                             new AccessRule(
                                     PATTERN_EVERYTHING,
                                     IAccessRule.K_DISCOURAGED | IAccessRule.IGNORE_IF_BETTER));
-                result.putIfAbsent(entry.getPath(), entry);
+                if (runtimeDependencyAvailable) {
+                    result.putIfAbsent(entry.getPath(), entry);
+                    unloadedEntries.putIfAbsent(entry.getPath(), null);
+                } else {
+                    unloadedEntries.putIfAbsent(entry.getPath(), entry);
+                    hasUnloadedEntries = true;
+                }
             }
         }
 
-        return result.values();
+        return new ClasspathHolder(
+                result.values(),
+                hasUnloadedEntries ? unloadedEntries.values() : Collections.emptyList());
+    }
+
+    private boolean entryAvailable(ClasspathEntry entry) throws CoreException {
```

Comment:
Need to come up with a better name, but this use to be in the Strategy, but moved it here to remove the list iteration that happened multiple times.